wpimath/src/main/java/edu/wpi/first/math/interpolation/InterpolatingDouble.java

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.math.interpolation;
+
+/**
+ * A Double that can be interpolated using the InterpolatingTreeMap.
+ *
+ * @see InterpolatingTreeMap
+ */
+public class InterpolatingDouble
+    implements Interpolatable<InterpolatingDouble>,
+        InverseInterpolable<InterpolatingDouble>,
+        Comparable<InterpolatingDouble> {
+  public Double value = 0.0;
+
+  public InterpolatingDouble(Double val) {
+    value = val;
+  }
+
+  @Override
+  public InterpolatingDouble interpolate(InterpolatingDouble other, double x) {
+    Double dydx = other.value - value;
+    Double searchY = dydx * x + value;
+    return new InterpolatingDouble(searchY);
+  }
+
+  @Override
+  public double inverseInterpolate(InterpolatingDouble upper, InterpolatingDouble query) {
+    double upper_to_lower = upper.value - value;
+    if (upper_to_lower <= 0) {
+      return 0;
+    }
+    double query_to_lower = query.value - value;
```

Comment:
This should be `queryToLower` instead to match WPILib's variable naming convention.